test/StringToolsTests.h

```diff
@@ -0,0 +1,333 @@
+    {//Character Checks
+        TEST_EQUAL("IsSpace(const_Char8)-Pass",true,StringTools::IsSpace(' '));
+        TEST_EQUAL("IsSpace(const_Char8)-Fail",false,StringTools::IsSpace('\t'));
+        TEST_EQUAL("IsTab(const_Char8)-Pass1",true,StringTools::IsTab('\t'));
+        TEST_EQUAL("IsTab(const_Char8)-Pass2",true,StringTools::IsTab('\v'));
+        TEST_EQUAL("IsTab(const_Char8)-Fail",false,StringTools::IsTab('\r'));
+        TEST_EQUAL("IsNewline(const_Char8)-Pass1",true,StringTools::IsNewline('\r'));
+        TEST_EQUAL("IsNewline(const_Char8)-Pass2",true,StringTools::IsNewline('\n'));
+        TEST_EQUAL("IsNewline(const_Char8)-Fail",false,StringTools::IsNewline('!'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass1",true,StringTools::IsWhitespace(' '));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass2",true,StringTools::IsWhitespace('\t'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Pass3",true,StringTools::IsWhitespace('\n'));
+        TEST_EQUAL("IsWhitespace(const_Char8)-Fail",false,StringTools::IsWhitespace('&'));
+        TEST_EQUAL("IsDigit(const_Char8)-Pass",true,StringTools::IsDigit('5'));
+        TEST_EQUAL("IsDigit(const_Char8)-Fail",false,StringTools::IsDigit('L'));
+        TEST_EQUAL("IsLowerAlphaLetter(const_Char8)-Pass",true,StringTools::IsLowerAlphaLetter('a'));
+        TEST_EQUAL("IsLowerAlphaLetter(const_Char8)-Fail",false,StringTools::IsLowerAlphaLetter('A'));
+        TEST_EQUAL("IsUpperAlphaLetter(const_Char8)-Pass",true,StringTools::IsUpperAlphaLetter('B'));
+        TEST_EQUAL("IsUpperAlphaLetter(const_Char8)-Fail",false,StringTools::IsUpperAlphaLetter('b'));
+        TEST_EQUAL("IsAlphaLetter(const_Char8)-Pass",true,StringTools::IsAlphaLetter('y'));
+        TEST_EQUAL("IsAlphaLetter(const_Char8)-Fail",false,StringTools::IsAlphaLetter('9'));
+        TEST_EQUAL("IsLowerHexLetter(const_Char8)-Pass",true,StringTools::IsLowerHexLetter('e'));
+        TEST_EQUAL("IsLowerHexLetter(const_Char8)-Fail",false,StringTools::IsLowerHexLetter('g'));
+        TEST_EQUAL("IsUpperHexLetter(const_Char8)-Pass",true,StringTools::IsUpperHexLetter('F'));
+        TEST_EQUAL("IsUpperHexLetter(const_Char8)-Fail",false,StringTools::IsUpperHexLetter('1'));
+        TEST_EQUAL("IsHexLetter(const_Char8)-Pass",true,StringTools::IsHexLetter('B'));
+        TEST_EQUAL("IsHexLetter(const_Char8)-Fail",false,StringTools::IsHexLetter('z'));
+        TEST_EQUAL("IsHexDigit(const_Char8)-Pass",true,StringTools::IsHexDigit('8'));
+        TEST_EQUAL("IsHexDigit(const_Char8)-Fail",false,StringTools::IsHexDigit('t'));
+        TEST_EQUAL("IsAlphanumeric(const_Char8)-Pass",true,StringTools::IsAlphanumeric('R'));
+        TEST_EQUAL("IsAlphanumeric(const_Char8)-Fail",false,StringTools::IsAlphanumeric(' '));
+    }//Character Manipulation and Checks
+
+    {//String Manipulation
+        String ToUpperString("This is typed in a reasonable, soft-spoken tone.");
+        String ToUpperResult("THIS IS TYPED IN A REASONABLE, SOFT-SPOKEN TONE.");
```

Comment:
/Nit Results for test should be const

Reply:
Not sure I got them all but I made a large number of the source strings const.